### ISSUE TITLE
Python Multiprocess Deadlock issue

### ISSUE THREAD

**Issue by maniraman-periyasamy** [Qiskit/qiskit-aer#1485]
### Informations

- **Qiskit Aer version**: 0.34.2
- **Python version**: 3.10.0
- **Operating system**: CentOS 7

### What is the current behavior?

Qiskit simulation never starts.
ProcessPoolExecutor threads run into deadlock.

### Steps to reproduce the problem

Run large enough simulations For ex. 22-25 Qubits Quantum Volume circuit with depth of 5

### What is the expected behavior?

Simulation to utilize all cores and run in parallel or run in a serial order at the least.

### Suggested solutions

Suggestion 1)
In parallel.py file under qiskit.tools
In line 69: 

elif sys.version_info[0] == 3 and sys.version_info[1] >= 8:
        PARALLEL_DEFAULT = False

Suggestion 2)

Also update the necessity of python 3.8 or lower in AerSimulator description and aer_simulator.py
(For past 2 years I have been scolding qiskit implementation for running in serial without knowing that the python 3.9 is to be downgraded!)

Thanks and regards,
Mani

**Comment by hhorii** [comment]
@maniraman-periyasamy 
Thank you for your reporting and I'm sorry that my response is lazy.
Currently Qiskit does not disable parallel processing in Linux environment for any versions of Python, as you pointed in `parallel.py`.

I set up CentOS 7 environment and tested QV. In my environment, no deadlock was happened. The test code is here:

```
from qiskit import QuantumCircuit, transpile
from qiskit.circuit.library import QuantumVolume
from qiskit.providers.aer import AerSimulator
circ = QuantumVolume(25)
circ.measure_all()
simulator = AerSimulator()
circ = transpile(circ, simulator)
result = simulator.run(circ).result()
print(result.get_counts())
``` 

My environment:
```
$ pip list | grep qiskit
qiskit               0.34.2
qiskit-aer           0.10.3
qiskit-ibmq-provider 0.18.3
qiskit-ignis         0.7.0
qiskit-terra         0.19.2

$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    1
Core(s) per socket:    8
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz
Stepping:              4
CPU MHz:               2294.561
BogoMIPS:              4589.25
Hypervisor vendor:     Xen
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              25344K
NUMA node0 CPU(s):     0-7
```

Do you still encoruneter deadlock in your environment?

**Comment by maniraman-periyasamy** [comment]
@hhorii 

Sorry, I should have been more specific.
This happens only when there are multiple sockets (and nodes). Only when I run the code in cluster.
The program works perfect in the local machine with single socket and CPU.

**Comment by hhorii** [comment]
can you tell more detail to reproduce your issue?
did you run python program via `ssh` or other remote shell?
did you run multiple python program in a single node?

**Comment by j7168908jx** [comment]
Just a note for others googling deadlock issue and found here,
I encountered deadlock using `aer-gpu` if I imported `multiprocessing` module even without using it.
The GPU can only run one instance and will become deadlocked for the second instance. 
Simple not using `multiprocessing`, and not importing it will make the GPU (I'm using single gpu) work.

I'm using Python 3.8.12 and
qiskit                            0.39.3
qiskit-aer                        0.11.1
qiskit-aer-gpu                    0.11.2
qiskit-ibmq-provider              0.19.2
qiskit-ignis                      0.7.1
qiskit-terra                      0.22.3

I know they are a bit old at this time, but just here to remind others if you encounter the same, since I haven't tested any newer version
